### PR TITLE
suppress sign conversion warning

### LINKED ISSUE #4: Build warnings (-Wsign-conversion) under Clang
MacOS Catalina 10.15.4 Beta
Apple clang version 11.0.0 (clang-1100.0.33.17)

```
--- stderr: spdlog_vendor                                                                                                
In file included from /opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/src/fmt.cpp:10:
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:484:48: warning: implicit conversion changes signedness: 'int' to 'unsigned long long' [-Wsign-conversion]
          (min_exponent + fp::significand_size - 1) * one_over_log2_10 +
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~  ~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:534:53: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
    auto result = static_cast<double_bigit>(bigits_[index]) - other - borrow;
                                            ~~~~~~~ ^~~~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:535:13: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
    bigits_[index] = static_cast<bigit>(result);
    ~~~~~~~ ^~~~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:541:38: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
    while (num_bigits > 0 && bigits_[num_bigits] == 0) --num_bigits;
                             ~~~~~~~ ^~~~~~~~~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:542:31: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
    bigits_.resize(num_bigits + 1);
            ~~~~~~ ~~~~~~~~~~~^~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:553:40: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
      subtract_bigits(i, other.bigits_[j], borrow);
                         ~~~~~         ^
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:605:25: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
      bigits_[num_bigits++] = n & ~bigit(0);
      ~~~~~~~ ~~~~~~~~~~^~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:608:20: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
    bigits_.resize(num_bigits);
            ~~~~~~ ^~~~~~~~~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:644:37: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
      bigit lhs_bigit = lhs.bigits_[i], rhs_bigit = rhs.bigits_[j];
                        ~~~         ^
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:644:65: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
      bigit lhs_bigit = lhs.bigits_[i], rhs_bigit = rhs.bigits_[j];
                                                    ~~~         ^
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:659:62: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
      return i >= n.exp_ && i < n.num_bigits() ? n.bigits_[i - n.exp_] : 0;
                                                 ~         ~~^~~~~~~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:699:20: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
    bigits_.resize(num_result_bigits);
            ~~~~~~ ^~~~~~~~~~~~~~~~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:707:44: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
        sum += static_cast<double_bigit>(n[i]) * n[j];
                                         ~ ^
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:707:52: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
        sum += static_cast<double_bigit>(n[i]) * n[j];
                                                 ~ ^
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:709:15: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
      bigits_[bigit_index] = static_cast<bigit>(sum);
      ~~~~~~~ ^~~~~~~~~~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:716:45: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
        sum += static_cast<double_bigit>(n[i++]) * n[j--];
                                         ~ ~^~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:716:55: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
        sum += static_cast<double_bigit>(n[i++]) * n[j--];
                                                   ~ ~^~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:717:15: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
      bigits_[bigit_index] = static_cast<bigit>(sum);
      ~~~~~~~ ^~~~~~~~~~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:735:33: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
      bigits_.resize(num_bigits + exp_difference);
              ~~~~~~ ~~~~~~~~~~~^~~~~~~~~~~~~~~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:737:17: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
        bigits_[j] = bigits_[i];
        ~~~~~~~ ^
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:737:30: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
        bigits_[j] = bigits_[i];
                     ~~~~~~~ ^
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:1031:18: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
      buf.resize(num_digits);
          ~~~~~~ ^~~~~~~~~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:1077:47: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
      while (num_digits > 0 && buf[num_digits - 1] == '0') {
                               ~~~ ~~~~~~~~~~~^~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:1182:30: warning: implicit conversion changes signedness: 'int' to 'size_t' (aka 'unsigned long') [-Wsign-conversion]
      std::memmove(p, p + 1, fraction_size);
      ~~~                    ^~~~~~~~~~~~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:1211:42: warning: implicit conversion changes signedness: 'int' to 'size_t' (aka 'unsigned long') [-Wsign-conversion]
      std::memmove(begin + 1, begin + 2, fraction_size);
      ~~~                                ^~~~~~~~~~~~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:1213:16: warning: implicit conversion changes signedness: 'int' to 'unsigned long' [-Wsign-conversion]
    buf.resize(fraction_size + offset + 1);
               ^~~~~~~~~~~~~ ~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:1182:30: warning: implicit conversion changes signedness: 'int' to 'size_t' (aka 'unsigned long') [-Wsign-conversion]
      std::memmove(p, p + 1, fraction_size);
      ~~~                    ^~~~~~~~~~~~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/src/fmt.cpp:156:40: note: in instantiation of function template specialization 'fmt::v6::internal::snprintf_float<double>' requested here
        template FMT_API int internal::snprintf_float(double, int,
                                       ^
In file included from /opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/src/fmt.cpp:10:
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:1211:42: warning: implicit conversion changes signedness: 'int' to 'size_t' (aka 'unsigned long') [-Wsign-conversion]
      std::memmove(begin + 1, begin + 2, fraction_size);
      ~~~                                ^~~~~~~~~~~~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:1213:16: warning: implicit conversion changes signedness: 'int' to 'unsigned long' [-Wsign-conversion]
    buf.resize(fraction_size + offset + 1);
               ^~~~~~~~~~~~~ ~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:1182:30: warning: implicit conversion changes signedness: 'int' to 'size_t' (aka 'unsigned long') [-Wsign-conversion]
      std::memmove(p, p + 1, fraction_size);
      ~~~                    ^~~~~~~~~~~~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/src/fmt.cpp:159:40: note: in instantiation of function template specialization 'fmt::v6::internal::snprintf_float<long double>' requested here
        template FMT_API int internal::snprintf_float(long double, int,
                                       ^
In file included from /opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/src/fmt.cpp:10:
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:1211:42: warning: implicit conversion changes signedness: 'int' to 'size_t' (aka 'unsigned long') [-Wsign-conversion]
      std::memmove(begin + 1, begin + 2, fraction_size);
      ~~~                                ^~~~~~~~~~~~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:1213:16: warning: implicit conversion changes signedness: 'int' to 'unsigned long' [-Wsign-conversion]
    buf.resize(fraction_size + offset + 1);
               ^~~~~~~~~~~~~ ~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:1077:47: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
      while (num_digits > 0 && buf[num_digits - 1] == '0') {
                               ~~~ ~~~~~~~~~~~^~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/src/fmt.cpp:162:40: note: in instantiation of function template specialization 'fmt::v6::internal::format_float<double>' requested here
        template FMT_API int internal::format_float(double, int, internal::float_specs,
                                       ^
In file included from /opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/src/fmt.cpp:10:
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:1077:47: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
      while (num_digits > 0 && buf[num_digits - 1] == '0') {
                               ~~~ ~~~~~~~~~~~^~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/src/fmt.cpp:164:40: note: in instantiation of function template specialization 'fmt::v6::internal::format_float<long double>' requested here
        template FMT_API int internal::format_float(long double, int,
                                       ^
In file included from /opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/src/fmt.cpp:10:
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:1031:18: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
      buf.resize(num_digits);
          ~~~~~~ ^~~~~~~~~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:1108:7: note: in instantiation of function template specialization 'fmt::v6::internal::fallback_format<double>' requested here
      fallback_format(value, buf, exp);
      ^
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:1031:18: warning: implicit conversion changes signedness: 'int' to 'std::size_t' (aka 'unsigned long') [-Wsign-conversion]
      buf.resize(num_digits);
          ~~~~~~ ^~~~~~~~~~
/opt/ros/master/build/spdlog_vendor/spdlog-1.5.0-prefix/src/spdlog-1.5.0/include/spdlog/fmt/bundled/format-inl.h:1108:7: note: in instantiation of function template specialization 'fmt::v6::internal::fallback_format<long double>' requested here
      fallback_format(value, buf, exp);
      ^
36 warnings generated.
---
Finished <<< spdlog_vendor [6min 45s]
```

### PR DESCRIPTION
**Pull request by Karsten1987**
closes #4 

I ran into the same problems as @rotu described in the referenced ticket.
Honestly, I did not install the package from `brew` as recommended and I totally agree that this problem should be fixed upstream. Likewise though, I feel that little fix here can be of temporary solution to continue building ROS2 cleanly even with `-Werror`. 

Signed-off-by: Karsten Knese <karsten@openrobotics.org>

### CODE REVIEW

**Review by dirk-thomas — APPROVED** [review]
With passing CI on all platforms.

**Comment by Karsten1987** [comment]
CI (`--packages-select spdlog_vendor`):
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9757/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5380/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7929/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9626/)

File: CMakeLists.txt [hunk 1]
```diff
@@ -45,7 +45,7 @@ macro(build_spdlog)
   else()
     list(APPEND extra_cmake_args "-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}")
   endif()
-  list(APPEND extra_cmake_args "-DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}")
+  list(APPEND extra_cmake_args "-DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS} -Wno-sign-conversion")
```
Comment (clalancette) [inline CMakeLists.txt]: I don't think this is going to work on Windows; you'll probably need an alternate complier flag there.

**Comment by clalancette** [comment]
> CI (`--packages-select spdlog_vendor`):

That doesn't work, unfortunately, since you still need ament and friends.  `--packages-up-to spdlog_vendor` should do it.

File: CMakeLists.txt [hunk 2]
```diff
@@ -49,6 +49,7 @@ macro(build_spdlog)
   list(APPEND extra_cmake_args "-DSPDLOG_BUILD_TESTS=OFF")
   list(APPEND extra_cmake_args "-DSPDLOG_BUILD_EXAMPLE=OFF")
   if(NOT WIN32)
+    list(APPEND extra_cmake_args "-DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS} -Wno-sign-conversion")
```
Comment (clalancette) [inline CMakeLists.txt:54]: Can you add a TODO comment here with a link the upstream `fmt` ticket?

**Comment by Karsten1987** [comment]
Thanks for the feedback.

CI seems happy so far.
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9767/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5387/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7936/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9632/)

**Review by clalancette — COMMENTED** [review]
Just one more comment, and this seems good to me.  I'd like to see green CI here too.

File: CMakeLists.txt (hunk 2 above)
Comment (Karsten1987) [reply CMakeLists.txt:54]: done in https://github.com/ros2/spdlog_vendor/pull/7/commits/aaf27f3bfd9075284424f6bc3916b5dade5bdeff

**Review by clalancette — APPROVED** [review]
Looks good, thanks for iterating.

**Comment by dirk-thomas** [comment]
The compiler warning is still present in the latest nightly build using `clang`: https://ci.ros2.org/view/nightly/job/nightly_linux_clang_libcxx/482/

See #15 for the follow up PR.

**Pull request merged**